lib/src/mock_document_reference.dart

```diff
@@ -103,7 +103,7 @@ class MockDocumentReference<T extends Object?> implements DocumentReference<T> {
         message: 'Some requested document was not found.',
       ));
     }
-    return _setRawData(data);
+    return _setRawData(Map.from(data));
```

Comment:
What is this for?

Reply:
Due to new support to FieldPath on **update**, the Map<String, dynamic> is not gonna cut it.. Seems this needs more work.